containers.go

```diff
@@ -972,3 +973,99 @@ func FindDockerImagesByTagPrefix(imageName string) ([]docker.APIImages, error) {
 
 	return matchingImages, nil
 }
+
+// SquashImage takes a docker image with multiple layers and squashes it into
+// a single layer.  SquashImage takes advantage of the fact that docker
+// squashes layers into a single later when exported from a container
+func SquashImage(ctx context.Context, repo, tag string) error {
+	client := DockerClient()
+	squashCtx, cancel := context.WithTimeout(context.Background(), time.Minute*30)
```

Comment:
Use the passed-in context rather than setting its own timeout. This gives the caller flexibility to set the limit they want.